Program.cs

```diff
@@ -313,6 +313,16 @@ static private void ExtractBulkExport(string zipPath, string bulkExportPath)
 
             // .NET 4.8 does not support the "overwriteFiles"-parameter. This is why we need to clear the folder before extracting the Zipfile into the Bulk export folder.
             ZipFile.ExtractToDirectory(zipPath, bulkExportPath);
+
+            //If the unpacked ZIP only contains a ZIP and the setting ExtractInnerZIP is set to true then extract that ZIP
+            if (ApplicationSettings.ExtractInnerZIP && Directory.GetFiles(bulkExportPath)[0].EndsWith(".zip")) 
+            {
+                string innerZIPPath = Directory.GetFiles(bulkExportPath)[0];
```

Comment:
There are a few things you should/could improve

- I would move this code to a sep function: ExtractFirstZipFileWhenPresent, it could take the bulkExportPath as parameter and return a bool if it did it, then you could log (not to proud of the function name) (if you do this a lot of the bullets below are very easy to fix)
- Pity that you are getting the files  twice 
- This  current code crashes if for some reason no files were in the directory and the special option is enabled
- It works even if there are multiple files in the directory, if by chance the first file has the zip extension (maybe you want to do something about that)
- You are logging a lot, but not the inner zip unzipping, I would add that to the emailbody